<?xml version="1.0" ?>
<sdf version="1.6">
  <model name="proximity_sick_wl9l-3">
    <link name="link">
      <sensor name="laser" type="ray">
        <plugin name="proximity_ray_plugin" filename="libROSProximityRayPlugin.so" >
          <output_state_topic>sensor_output</output_state_topic>
          <time_delay>0.0005</time_delay><!-- this modifies the update rate of the ray sensor -->
          <output_function>normally_closed</output_function> <!-- energized when target detected -->
        </plugin>
        <ray>
          <range>
            <min>0.0</min>
            <max>1.0</max> <!-- this represents the distance at which the reflector's placed. max 12m -->
            <resolution>0.001</resolution>
          </range>

          <!-- nothing below this line really needs to be modified upon change of sensor model -->

          <scan>
            <horizontal>
              <samples>1</samples>
              <resolution>1</resolution>
              <min_angle>0</min_angle>
              <max_angle>0</max_angle>
            </horizontal>
          </scan>
          <noise>
            <type>gaussian</type>
            <mean>0.0</mean>
            <stddev>0.01</stddev>
          </noise>
        </ray>

        <always_on>1</always_on>
        <visualize>true</visualize>
      </sensor>
      <gravity>false</gravity>
      <inertial>
        <mass>0.1</mass>
      </inertial>
      <visual name="visual">
        <geometry>
          <mesh>
            <uri>model://hokuyo/meshes/hokuyo.dae</uri>
          </mesh>
        </geometry>
      </visual>
    </link>
  </model>
</sdf>
